homeassistant/components/homekit_controller/__init__.py

```diff
@@ -101,15 +85,16 @@ def accessory_setup(self):
                 self.hass, RETRY_INTERVAL, lambda _: self.accessory_setup())
             return
         for accessory in data:
-            serial = get_serial(accessory)
-            if serial in self.hass.data[KNOWN_ACCESSORIES]:
-                continue
-            self.hass.data[KNOWN_ACCESSORIES][serial] = self
             aid = accessory['aid']
             for service in accessory['services']:
+                iid = service['iid']
+                if (aid, iid) in self.entities:
```

Comment:
Where are these ids added to the list?

Reply:
Good spot. This code doesn't even exist anymore in the config flow branch, the append happens in a function that doesn't exist yet by `async_setup_entry`. Doh. PR incoming.